src/ui/map.js

```diff
@@ -2174,7 +2177,13 @@ class Map extends Camera {
 
     // NOTE: Make fog non-optional
     getFog(): Fog | null {
-        return this.style.getFog();
+        return this.style ? this.style.getFog() : null;
+    }
+
+    getFogOpacity(lnglat: LngLatLike): number {
+        return this.style && this.style.fog ?
```

Comment:
Sometimes I like to transform this sort of thing into an early return for clarity, but no change recommended.

```suggestion
        if (!this.style || !this.style.fog) return 0.0;
        return this.style.fog.sampler.getFogOpacityAtLatLng(LngLat.convert(lnglat), this.transform)
```